### PR TITLE
Sync local current user data on MainActivity create / resume

### PR DESCRIPTION
**Pull request by luk1337**
# AniTrend Pull Request

Thank you for contributing! Please take a moment to review our [**contributing guidelines**](https://github.com/AniTrend/anitrend-app/blob/master/CONTRIBUTING.md)
to make the process easy and effective for everyone involved.

**Please open an issue** before embarking on any significant pull request, especially those that
add a new library or change existing tests, otherwise you risk spending a lot of time working
on something that might not end up being merged into the project.

Before opening a pull request, please ensure:

- [x] You have followed our [**contributing guidelines**](https://github.com/AniTrend/anitrend-app/blob/master/CONTRIBUTING.md)
- [x] double-check your branch is based on `develop` and targets `develop` 
- [ ] Pull request has tests
- [x] Code is well-commented, linted and follows project conventions
- [x] Documentation is updated (if necessary)
- [x] Description explains the issue/use-case resolved
- [x] I did not commit files that are excluded in the .gitignore file (Happens if you stage files with Android Studio)

## Description
Previously we only updated the presenter current user on logging in, this makes it do that on MainActivity create / resume.

## Motivation and Context
See description.

## How Has This Been Tested?
Open app, change avatar/cover photo, close and reopen app.

## Screenshots (if appropriate):
n/a

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

**IMPORTANT**: By submitting a patch, you agree to allow the project
owners to license your work under the terms of the [MIT License](https://github.com/AniTrend/anitrend-app/blob/master/LICENSE.md).

### CODE REVIEW

File: app/src/main/java/com/mxt/anitrend/view/activity/index/MainActivity.kt [hunk 1]
```diff
@@ -240,6 +243,7 @@ class MainActivity : ActivityBase<Void, BasePresenter>(), View.OnClickListener,
         mDrawerLayout.addDrawerListener(mDrawerToggle)
         mDrawerToggle.syncState()
         updateUI()
+        requestCurrentUser()
```
Comment (wax911) [inline app/src/main/java/com/mxt/anitrend/view/activity/index/MainActivity.kt:246]: `onSaveInstanceState` is usually called when this activity is about to get paused, suggesting a user is leaving the screen thus calling `requestCurrentUser` at this point would do the opposite of the **create/resume** behaviour you're after. Did you mean to have this run in `onResume`?

Comment (luk1337) [reply app/src/main/java/com/mxt/anitrend/view/activity/index/MainActivity.kt:246]: This is onResume().
image

**Comment by wax911** [comment]
Major thanks @luk1337 ❤️ I had plans to have run as recurring task with a worker in the future but this does a good already so that'd be for another day

**Comment by luk1337** [comment]
Technically it'd be possible to do it from notification worker too so that it'd update UI live but that's kinda not worth it IMO.

**Pull request merged**